torchgeo/datasets/bright.py

```diff
@@ -0,0 +1,338 @@
+
+    colormap = (
+        'white',  # background
+        'green',  # intact
+        'burlywood',  # damaged
+        'red',  # destroyed
+    )
+
+    md5 = '2c435bb50345d425390eff59a92134ac'
+
+    url = 'https://huggingface.co/datasets/Kullervo/BRIGHT/resolve/50901f05db4acbd141e7c96d719d8317910498fb/dfc25_track2_trainval.zip'
+
+    data_dir = 'dfc25_track2_trainval'
+
+    valid_split = ('train', 'val', 'test')
+
+    # train_setlevels.txt are the training samples
+    # holdout_setlevels.txt are the validation samples
+    # val_setlevels.txt are the test samples
+    split_files: ClassVar[dict[str, str]] = {
+        'train': 'train_setlevel.txt',
+        'val': 'holdout_setlevel.txt',
+        'test': 'val_setlevel.txt',
+    }
+
+    px_class_values: ClassVar[dict[int, str]] = {
+        0: 'background',
+        1: 'intact',
+        2: 'damaged',
+        3: 'destroyed',
+    }
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new BRIGHT DFC2025 dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: train/val/test split to load
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+            AssertionError: If *split* is not one of 'train', 'val', or 'test.
+        """
+        assert split in self.valid_split, f'Split must be one of {self.valid_split}'
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        self.sample_paths = self._get_paths()
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and target at that index
+        """
+        idx_paths = self.sample_paths[index]
+
+        pre_image = self._load_image(idx_paths['pre_image']).float()
+        post_image = self._load_image(idx_paths['post_image']).float()
+        # https://github.com/ChenHongruixuan/BRIGHT/blob/11b1ffafa4d30d2df2081189b56864b0de4e3ed7/dfc25_benchmark/dataset/make_data_loader.py#L101
+        # post image is stacked to also have 3 channels
+        post_image = repeat(post_image, 'c h w -> (repeat c) h w', repeat=3)
+
+        sample = {'pre_image': pre_image, 'post_image': post_image}
```

Comment:
Not sure if we've finalized a standard for CD datasets yet. I think we're currently leaning towards T x C x H x W.

Reply:
https://github.com/microsoft/torchgeo/blob/8a3b60df4e82877e46ea3cd7fb97636149f0f980/torchgeo/datasets/levircd.py#L70 Levir Change detection does separate dict entries, I think 'pre_image' and 'post_image' is more clear than 'image1' and 'image2'

Reply:
We're going to reformat Levir soon. Note that Kornia AugmentationSequential only supports "image*`, not "*image".